### ISSUE TITLE
Should this work with the server behind a NAT? 

### ISSUE THREAD

**Issue by trankin** [deepch/RTSPtoWeb#138]
I have this same issue opened on the RTSPtoWebRTC project, but since it's being recommended to migrate to this project I thought I might try to resurface it here.

RTSPtoWeb is working flawlessly for me when on a local network, but I cannot get it to work as expected when the viewer is on another network.

I'm under the impression that the STUN servers should help both sides navigate the NAT and establish a connection.

I've exposed the RTSPtoWeb HTTP port via ngrok to reverse proxy to the server, the client does get signaling information just fine, but fails to actually connect.

I can do other WebRTC tests just fine from the same computer that is running RTSPtoWeb, but with the built in camera, and they work just fine. There may be (likely are) some complexities with with establishing the peer to peer connection that I don't understand, but this case that I'm exploring is one that I perceive should generally be possible, and my understanding is that a TURN server is needed for some rarer edge cases.

client browser -> firewall/router (nat) -> internet <- firewall/router (nat) (port forwarding 8083) <- server (RTSPToWebRTC) <- IP Camera

My overall goal is the above setup, but without port forwarding the signaling port and using something like mqtt to kick off the nat traversal.

Another note.  I can see that @chriswiggins has had a PR merged that includes the ability to set STUN configuration, and I can confirm that I'm now seeing srflx entries in my remote description, I had a lot of hope that it would resolve this issue but it appears that there is still something missing.

**Comment by allenporter** [comment]
I got this working when configuring the stun server, and properly configuring ice candidates in the SDP offer.  See the discussion in https://github.com/home-assistant/frontend/pull/12084

**Comment by trankin** [comment]
@allenporter, thank you very much for this (I was able to successfully test with this direction).  If I interpret this correctly, this basically means that RTSPtoWeb isn't handling trickleice?

**Comment by allenporter** [comment]
RTSPtoWeb does not have a way to send ice candidates in the signal path that I am aware of, at the moment. Instead, they must all be sent in the offer. When configuring the viewer side offer with ice candidates it works for me.

**Comment by hotkhwan** [comment]
Hi share my scenario now is work on public IP for the benefit of others 
1 config.json 
    "http_port": ":8083",   # allow web access and API
    "webrtc_port_max": 61000,
    "webrtc_port_min": 60000
    allow allowing UDP connection port 
two method 
2.1 If using one-by-one NAT allow policy firewall port TCP 8083 and UDP 60000 to 61000
2.2 If using port forwarding mute be same IP incoming and outgoing  and forward TCP 8083, UDP 60000 to 61000 and allow firewall policy <check what is my IP web site mute be name public IP>

**Comment by Abdeladim101Basraoui** [comment]
@hotkhwan  could you please explain the two methods ?

**Comment by hotkhwan** [comment]
> @hotkhwan,@trankin could you please explain the two methods ?
@Abdeladim101Basraoui  Example diagram my scenario 
![WebRTCNAT](https://user-images.githubusercontent.com/13896960/194352169-933bcbeb-44cb-45e1-8d3f-aa3cffbc103e.png)

**Comment by Abdeladim101Basraoui** [comment]
@hotkhwan thanks for quick responding, I'm a beginner with this solution if I quite understand that the RTSPtowebRTC should run on a LAN network with the RTSP Ip cam and allow the port 8083 on the firewall for TCP streaming ,and on the other side get the feed with the WAN ip address, for the record that works good on local but ,my issue is when I try to deploy the app to use it no feed i coming ,and with a time out message

if you could help that sounds fantastic beside that ,thanks @hotkhwan for this explication

**Comment by teleoltre** [comment]
@hotkhwan Sorry.
Where can I find config.json file on my home assistant? Thank you

**Comment by allenporter** [comment]
If you are using home assistant, the feature you want is in the next release which has the ability to add a stun server in the JavaScript/frontend.

https://github.com/allenporter/stream-addons/issues/62

**Comment by teleoltre** [comment]
@allenporter Thanks for the answer, I have noticed that my problem is the pfsense firewall.
- If they are local on the same Vlan everything works fine.
- If I am on a different vlan and I set a rule on the firewall to forward all traffic between the two vlan and the two devices (client and HA), everything works fine.

But if I try to connect remotely, even using a VPN, cloudflare or nabu casa, it only shows me a frame but the camera is still

I can't find a solution to this.
